tests/AbstractLexerTest.php

```diff
@@ -279,4 +282,60 @@ public function testMarkerAnnotationLocaleTr(): void
         self::assertEquals('@', $mutableLexer->token->value);
         self::assertEquals('ODM\Id', $mutableLexer->lookahead->value);
     }
+
+    public function testCanTokenizeFloatValue(): void
+    {
+        $lexer = new /** @template-extends AbstractLexer<int, string|int|float|bool> */ class () extends AbstractLexer {
+            public const T_NONE    = 1;
+            public const T_INTEGER = 2;
+            public const T_FLOAT   = 4;
+            public const T_BOOL    = 4;
+
+            protected function getType(string|int|float|bool &$value): int
+            {
+                if ($value === 'y') {
+                    $value = true;
+
+                    return self::T_BOOL;
+                }
+
+                if (is_numeric($value)) {
+                    $value += 0;
```

Comment:
In the case of this parser, is there actually a point in making `$value` something else than `string` or `int`? Like… do we want the token to be built with `float` or `bool` rather than `string` or `int`? Or would you say it's just an accidental consequence of doing this `+=` operation? Cause if yes, a simple workaround would be to create another variable and use it in the condition below, instead of altering `$value`.

Reply:
I didn't write the original code, so I can only guess. But it feels like it is intentionally casting to `int|float` via the `+=` trick. And that `int|float` value will later be used as-is in the code using the parser. I _think_ it makes sense that a parser is able to return "real" values and not strings that still need to be cast to something.

Not sure, if you are familiar with `creo/wkt-parser`, but they are parsing strings like `"POINT(12.45 0.4)"`, hence the need to have real numeric values at some point.

Does that answer your question ?

Reply:
I'm not familiar with this package, but yes, it does answer the question 👍